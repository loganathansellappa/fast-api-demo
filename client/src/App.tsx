import {useRef} from "react";
import {serverData} from "./utils/HelperUtils.ts";
import {createMissionControlApiClient} from "./services/missionControlClient.ts";
import {AxiosClientProvider} from "./context/AxiosClientContext.ts";
import {QueryClientProvider} from "react-query";
import {MissionControlList} from "./components/MissionControlList/MissionControlList.tsx";

export const App = () => {
	const {
		current: {
			axiosClientService,
			queryClient,
		},
	} = useRef(
		createMissionControlApiClient({
			endpoint: serverData().baseUrl,
		}),
	)

	return (
		<AxiosClientProvider value={axiosClientService}>
			<QueryClientProvider client={queryClient}>
				<MissionControlList />
			</QueryClientProvider>
		</AxiosClientProvider>
	);
};

