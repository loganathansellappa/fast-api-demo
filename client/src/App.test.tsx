import { render } from "@testing-library/react";
import { App } from './App.tsx';

let element: { container: HTMLElement };

test("renders App component", () => {
  element = render(<App />);
  expect(element.container.firstChild).toMatchSnapshot();
  expect(element.container.textContent).toContain("Flight Mission ControlAdd Mission Control");
});