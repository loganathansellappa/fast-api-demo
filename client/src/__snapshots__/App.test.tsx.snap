// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div
  class="w-full flex flex-col h-screen bg-gradient-to-b from-gray-100 to-gray-300 p-10"
>
  <div
    class="flex"
  >
    <h1
      class="bg-gray-100 text-black font-bold py-2 px-4 rounded"
    >
      Flight Mission Control
    </h1>
    <button
      class="bg-gray-200 hover:bg-gray-300 text-black font-bold py-2 px-4 rounded ml-auto"
    >
      Add Mission Control
    </button>
  </div>
  <div
    class="flex flex-col items-center justify-center h-screen"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      Request is processing!
    </h1>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
